test_socket.py

```diff
@@ -0,0 +1,232 @@
+import socket
+import string
+import email.utils
+
+
+def test_text():
+    ADDR = ('127.0.0.1', 10007)
+    client = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+    )
+    client.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    client.connect(ADDR)
+    msg = "GET sample.txt HTTP/1.1"+"\r\n"+"Host: www.example.com"+"\r\n\r\n"
+    client.sendall(msg)
+    client.shutdown(socket.SHUT_WR)
+    part = ""
+    while True:
+        tempPart = client.recv(16)
+        part = part + tempPart
+        if len(tempPart) < 16:
+            client.close()
+            break
+    part_list = string.split(part, '\r\n')
+    assert part_list[0].find("200 OK") != -1
+    date = email.utils.formatdate(usegmt=True)
+    assert part_list[1].find(date) != -1
+    assert part_list[2].find("Content-Type: text/plain") != -1
+    assert part_list[3].find("Content-Length:") != -1
+    assert part_list[5].find("This is a very simple text file") != -1
+
+
+def test_directory():
+    ADDR = ('127.0.0.1', 10007)
+    client = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+    )
+    client.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    client.connect(ADDR)
+    msg = "GET images HTTP/1.1"+"\r\n"+"Host: www.example.com"+"\r\n\r\n"
+    client.sendall(msg)
+    client.shutdown(socket.SHUT_WR)
+    part = ""
+    while True:
+        tempPart = client.recv(16)
+        part = part + tempPart
+        if len(tempPart) < 16:
+            client.close()
+            break
+    part_list = string.split(part, '\r\n')
+    assert part_list[0].find("200 OK") != -1
+    date = email.utils.formatdate(usegmt=True)
+    assert part_list[1].find(date) != -1
+    assert part_list[2].find("Content-Type: text/html") != -1
+    assert part_list[3].find("Content-Length:") != -1
+    assert part_list[5].find("JPEG_example.jpg") != -1
```

Comment:
These are not unreasonable tests for your http responses.  You are getting a lot closer to a 'correct' test.  You are checking headers and the first line and so on, splitting on CRLF so that you validate that that separator is used.  But there's still a few issues.  

Headers do not have to be in any particular order.  Most systems implement headers as a dictionary internally.  This means that you are relying on an implementation detail of your response to validate your response.  That's not so great.